File: src/main/java/org/model/RendezVous/postulation.java
package org.model.RendezVous;

import java.time.LocalDateTime;

public class postulation {

    private int id_postulation;
    private int id_rendez_vous;
    private int id_postulant;

    private String mail_user;
    private String message;
    private String cv;

    private String status; // en_attente, accepte, refuse
    private LocalDateTime date_postulation;

    private String role;   // freelance, entrepreneur
    private String nom;
    private String prenom;
    private String titre;


    // ================= CONSTRUCTEUR VIDE =================
    public postulation() {}

    // ================= CONSTRUCTEUR COMPLET =================
    public postulation(int id_postulation,
                       int id_rendez_vous,
                       int id_postulant,
                       String mail_user,
                       String message,
                       String cv,
                       String status,
                       LocalDateTime date_postulation,
                       String role,
                       String nom,
                       String prenom) {

        this.id_postulation = id_postulation;
        this.id_rendez_vous = id_rendez_vous;
        this.id_postulant = id_postulant;
        this.mail_user = mail_user;
        this.message = message;
        this.cv = cv;
        this.status = status;
        this.date_postulation = date_postulation;
        this.role = role;
        this.nom = nom;
        this.prenom = prenom;
    }

    // ================= GETTERS & SETTERS =================

    public int getId_postulation() {
        return id_postulation;
    }

    public void setId_postulation(int id_postulation) {
        this.id_postulation = id_postulation;
    }

    public int getId_rendez_vous() {
        return id_rendez_vous;
    }

    public void setId_rendez_vous(int id_rendez_vous) {
        this.id_rendez_vous = id_rendez_vous;
    }

    public int getId_postulant() {
        return id_postulant;
    }

    public void setId_postulant(int id_postulant) {
        this.id_postulant = id_postulant;
    }

    public String getMail_user() {
        return mail_user;
    }

    public void setMail_user(String mail_user) {
        this.mail_user = mail_user;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getCv() {
        return cv;
    }

    public void setCv(String cv) {
        this.cv = cv;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public LocalDateTime getDate_postulation() {
        return date_postulation;
    }

    public void setDate_postulation(LocalDateTime date_postulation) {
        this.date_postulation = date_postulation;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }
    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    // ================= TO STRING =================
    @Override
    public String toString() {
        return "Postulation{" +
                "id=" + id_postulation +
                ", nom='" + nom + " " + prenom + '\'' +
                ", status='" + status + '\'' +
                '}';
    }
}

File: src_backup_phpstan/Entity/RendezVous/Postulation.php
<?php

namespace App\Entity\RendezVous;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'postulation')]
class Postulation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'id_postulation', type: 'integer')]
    private ?int $idPostulation = null;

    #[ORM\Column(name: 'id_rendez_vous', type: 'integer')]
    private ?int $idRendezVous = null;

    #[ORM\Column(name: 'id_postulant', type: 'integer')]
    private ?int $idPostulant = null;

    #[ORM\Column(name: 'mail_user', type: 'string', length: 150, nullable: true)]
    private ?string $mailUser = null;

    #[ORM\Column(name: 'message', type: 'text', nullable: true)]
    private ?string $message = null;

    #[ORM\Column(name: 'cv', type: 'string', length: 255, nullable: true)]
    private ?string $cv = null;

    #[ORM\Column(name: 'status', type: 'string', length: 50)]
    private ?string $status = 'en_attente';

    #[ORM\Column(name: 'date_postulation', type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $datePostulation = null;

    #[ORM\Column(name: 'role', type: 'string', length: 50, nullable: true)]
    private ?string $role = null;

    #[ORM\Column(name: 'nom', type: 'string', length: 100, nullable: true)]
    private ?string $nom = null;

    #[ORM\Column(name: 'prenom', type: 'string', length: 100, nullable: true)]
    private ?string $prenom = null;

    #[ORM\Column(name: 'titre', type: 'string', length: 200, nullable: true)]
    private ?string $titre = null;
    

    public function getIdPostulation(): ?int
    {
        return $this->idPostulation;
    }

    public function getIdRendezVous(): ?int
    {
        return $this->idRendezVous;
    }

    public function setIdRendezVous(?int $idRendezVous): self
    {
        $this->idRendezVous = $idRendezVous;
        return $this;
    }

    public function getIdPostulant(): ?int
    {
        return $this->idPostulant;
    }

    public function setIdPostulant(?int $idPostulant): self
    {
        $this->idPostulant = $idPostulant;
        return $this;
    }

    public function getMailUser(): ?string
    {
        return $this->mailUser;
    }

    public function setMailUser(?string $mailUser): self
    {
        $this->mailUser = $mailUser;
        return $this;
    }

    public function getMessage(): ?string
    {
        return $this->message;
    }

    public function setMessage(?string $message): self
    {
        $this->message = $message;
        return $this;
    }

    public function getCv(): ?string
    {
        return $this->cv;
    }

    public function setCv(?string $cv): self
    {
        $this->cv = $cv;
        return $this;
    }

    public function getStatus(): ?string
    {
        return $this->status;
    }

    public function setStatus(?string $status): self
    {
        $this->status = $status;
        return $this;
    }

    public function getDatePostulation(): ?\DateTimeInterface
    {
        return $this->datePostulation;
    }

    public function setDatePostulation(?\DateTimeInterface $datePostulation): self
    {
        $this->datePostulation = $datePostulation;
        return $this;
    }

    public function getRole(): ?string
    {
        return $this->role;
    }

    public function setRole(?string $role): self
    {
        $this->role = $role;
        return $this;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(?string $nom): self
    {
        $this->nom = $nom;
        return $this;
    }

    public function getPrenom(): ?string
    {
        return $this->prenom;
    }

    public function setPrenom(?string $prenom): self
    {
        $this->prenom = $prenom;
        return $this;
    }

    public function getTitre(): ?string
    {
        return $this->titre;
    }

    public function setTitre(?string $titre): self
    {
        $this->titre = $titre;
        return $this;
    }
}